app/controllers/projects/samples_controller.rb

```diff
@@ -149,22 +150,42 @@ def query
       authorize! @project, to: :sample_listing?
 
       @search_params = search_params
-      set_metadata_fields
+
+      metadata_fields(@search_params['metadata_template'])
+
       advanced_search_fields(@project.namespace)
 
-      @query = Sample::Query.new(@search_params.except(:metadata).merge({ project_ids: [@project.id] }))
+      @query = Sample::Query.new(@search_params.except('metadata_template').merge({ project_ids: [@project.id] }))
     end
 
     def search_params
       updated_params = update_store(search_key,
                                     params[:q].present? ? params[:q].to_unsafe_h : {}).with_indifferent_access
+      updated_params.slice!(:name_or_puid_cont, :name_or_puid_in, :groups_attributes, :metadata_template, :sort)
 
       if !updated_params.key?(:sort) ||
-         (updated_params[:metadata].to_i.zero? && updated_params[:sort]&.match?(/metadata_/))
+         (updated_params[:metadata_template] == 'none' && updated_params[:sort]&.match?(/metadata_/))
         updated_params[:sort] = 'updated_at desc'
         update_store(search_key, updated_params)
       end
       updated_params
     end
+
+    def current_metadata_template
```

Comment:
Since this method is in this controller as well as the groups samples controller, lets move this method into the metadata concern

Reply:
Updated in a75618a